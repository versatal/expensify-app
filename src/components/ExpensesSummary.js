import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectTotal from '../selectors/expense-total';

export const ExpensesSummary = ( {expenseCount, expensesTotal} ) => (
  <div>
    {expenseCount === 1  
      ? <p>{`Viewing ${expenseCount} expense totalling ${numeral(expensesTotal / 100).format('$0,0.00')}`}</p> 
      : <p>{`Viewing ${expenseCount} expenses totalling ${numeral(expensesTotal / 100).format('$0,0.00')}`}</p>}
  </div>
);

const mapStateToProps = (state) => {
  return {
    expenseCount: selectExpenses(state.expenses, state.filters).length,
    expensesTotal: selectTotal(selectExpenses(state.expenses, state.filters))
  }
};

export default connect(mapStateToProps)(ExpensesSummary);