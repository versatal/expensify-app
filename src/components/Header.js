import React from 'react';
import { NavLink } from 'react-router-dom'

const Header = () => (
  <header>
    <h1>Expensify</h1>
    <NavLink to="/" activeClassName="isActive" exact={true}>Dashboard</NavLink>
    <NavLink to="/create" activeClassName="isActive">Create Expense</NavLink>
  </header>
)

export default Header;