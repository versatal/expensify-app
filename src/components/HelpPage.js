import React from 'react';

const HelpPage = () => (
  <div>
    This is from help page component
  </div>
);

export default HelpPage;