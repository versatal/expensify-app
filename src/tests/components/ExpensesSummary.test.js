import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render summary phrase correctly with one expense', () => {
  const expenseCount = 1;
  const expensesTotal = 9434;
  const wrapper = shallow(<ExpensesSummary expenseCount={expenseCount} expensesTotal={expensesTotal}/>);
  expect(wrapper).toMatchSnapshot();
});

test('should render summary phrase correctly with multiple expenses', () => {
  const expenseCount = 2;
  const expensesTotal = 9434;
  const wrapper = shallow(<ExpensesSummary expenseCount={expenseCount} expensesTotal={expensesTotal}/>);
  expect(wrapper).toMatchSnapshot();
});