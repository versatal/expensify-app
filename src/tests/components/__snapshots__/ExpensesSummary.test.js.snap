// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render summary phrase correctly with multiple expenses 1`] = `
<div>
  <p>
    Viewing 2 expenses totalling $94.34
  </p>
</div>
`;

exports[`should render summary phrase correctly with one expense 1`] = `
<div>
  <p>
    Viewing 1 expense totalling $94.34
  </p>
</div>
`;
